3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -118,31 +70,28 @@ function add_member() {
   local replset_addr
   replset_addr="${MONGODB_REPLICA_NAME}/${endpoints}"
 
-  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${script}" --quiet; then
+  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "while (!rs.add('${host}').ok) { sleep(100); }" --quiet; then
```

Comment:
I don't think that's true, or I don't understand your point.  With stateful sets, the replica DNS name is stable, not the IP address, therefore we are using the DNS name here.